.github/workflows/publish-docs.yaml

```diff
@@ -0,0 +1,22 @@
+name: Publish Documentation
+on:
+  push:
+    branches: [ documentation/S3CSI-17-setup-github-pages ]
```

Comment:
This will be changed to `main` before merging.
Using current branch for reviewers